examples/ecies-example.cpp

```diff
@@ -0,0 +1,230 @@
+     * - HMAC(SHA512)
+     * - AES CBC with PKCS padding, zero IV and 256 bit key size
+     * - empty salts for KDF and MAC
+     */
+    const auto digest = DigestTypes::SHA512;
+    const int digestLength = Hash::getDigestSize(digest);
+    const auto kdfFunc = std::make_shared<X963KDF>(digest);
+    /* Dependency injection function, as the key for MAC is generated based on a random ECC key in the ECIES class */
+    const auto macFunc = std::function<std::unique_ptr<MessageAuthenticationCode>(const std::vector<uint8_t>&)>(
+                [digest](const std::vector<uint8_t> &key) -> std::unique_ptr<MessageAuthenticationCode> {
+        return std::make_unique<mococrw::HMAC>(digest, key);
+    });
+
+    /* The symmetric cipher */
+    const SymmetricCipherMode mode = SymmetricCipherMode::CBC;
+    const SymmetricCipherKeySize keySize = SymmetricCipherKeySize::S_256;
+    const SymmetricCipherPadding padding = SymmetricCipherPadding::PKCS;
+    const std::vector<uint8_t> zeroIV = std::vector<uint8_t>(AESCipherBuilder::getDefaultIVLength(mode));
+    const auto aesFunc = [zeroIV](const std::vector<uint8_t> &key) -> std::unique_ptr<SymmetricCipherI> {
+                    return AESCipherBuilder(mode, keySize, key).setIV(zeroIV).setPadding(padding).buildEncryptor();
+                };
+
+    /* Read the public key */
+    auto pubKey = AsymmetricPublicKey::readPublicKeyFromPEM(pubKeyPem);
+
+    /* Get the encryption context */
+    ECIESCtxBuilder encBuilder;
+    std::unique_ptr<ECIESEncryptionCtx> encCtx = encBuilder
+            // This is optional. The default is X963 with SHA512
+            .setKDF(kdfFunc)
+            // This is optional. The default is HMAC with SHA512
+            .setMacFactoryFunction(macFunc)
+            // This is optional. The default key length is 512 / 8 bytes (length of hash sum)
+            .setMacKeySize(digestLength)
+            // The next two lines are optional. Default: AES CBC with PKCS padding, zero IV and 256 bit key size
+            .setSymmetricCipherFactoryFunction(aesFunc)
+            .setSymmetricCipherKeySize(getSymmetricCipherKeySize(keySize))
+            .buildEncryptionCtx(pubKey);
+
+    std::vector<uint8_t> encryptedData;
+    try {
+        encCtx->update(message);
+        encryptedData = encCtx->finish();
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Error encrypting data." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    return EciesEncryptResult{
+        /* The serialization of the ephemeral key's public component is up to the implementer. The standard is not
+         * defining a format. Available formats in MoCOCrW are:
+         * - uncompressed (used here)
+         * - compressed
+         * - hybrid
+         */
+        encCtx->getEphemeralKey().toECPoint(openssl::EllipticCurvePointConversionForm::uncompressed),
+                encryptedData,
+                encCtx->getMAC(),
```

Comment:
I find the indentation a bit unexpected. Any reason the three members of the struct are not indented to the same column?

```suggestion
        encryptedData,
        encCtx->getMAC(),
```

Reply:
It's the default of my IDE.
It's fixed now.